lib/livebook_web/live/session_live.ex

```diff
@@ -1951,4 +1988,12 @@ defmodule LivebookWeb.SessionLive do
 
     :ok
   end
+
+  defp session_only_secrets(secrets, notebook_secrets) do
+    Enum.reject(secrets, &(&1.name in get_in(notebook_secrets, [Access.all(), "name"])))
+  end
+
+  defp is_secret_on_session?(secret_name, secrets) do
+    secret_name in Enum.map(secrets, & &1.name)
```

Comment:
We actually need to check if the value matches as well. i.e. if there is a secret in the session with the same name and different value, then it is another secret.